src/main/java/com/jsoniter/IterImplForStreaming.java

```diff
@@ -487,48 +487,56 @@ public final static int readStringSlowPath(JsonIterator iter, int j) throws IOEx
         }
     }
 
-    static long readLongSlowPath(JsonIterator iter, long value) throws IOException {
+    static long readLongSlowPath(final JsonIterator iter, long value, final boolean negative) throws IOException {
+        value = -value; // add negatives to avoid redundant checks for Long.MIN_VALUE on each iteration
+        long limit = negative ? Long.MIN_VALUE : -Long.MAX_VALUE;
+        long multmin = -922337203685477580L; // limit / 10
         for (; ; ) {
             for (int i = iter.head; i < iter.tail; i++) {
                 int ind = IterImplNumber.intDigits[iter.buf[i]];
                 if (ind == IterImplNumber.INVALID_CHAR_FOR_NUMBER) {
                     iter.head = i;
-                    return value;
+                    return negative ? value : -value;
                 }
-                if (value == Long.MIN_VALUE) {
-                    throw iter.reportError("readLongSlowPath", "value is too large for long");
+                if (value < multmin) {
+                    throw iter.reportError("readIntSlowPath", "value is too large for int");
```

Comment:
`throw iter.reportError("readIntSlowPath", "value is too large for int");`

should be long

Reply:
fixed by following 2 commits

please sqash them in case of merge